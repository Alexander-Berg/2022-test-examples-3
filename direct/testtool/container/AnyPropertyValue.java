package ru.yandex.direct.internaltools.tools.testtool.container;

import javax.annotation.ParametersAreNonnullByDefault;

import ru.yandex.direct.internaltools.core.annotations.input.Input;
import ru.yandex.direct.internaltools.core.annotations.input.Select;
import ru.yandex.direct.internaltools.core.container.InternalToolParameter;
import ru.yandex.direct.internaltools.tools.testtool.preprocessors.SetAnyPropertyToolPreProcessor;

@ParametersAreNonnullByDefault
public class AnyPropertyValue extends InternalToolParameter {
    @Input(label = "Имя проперти", processors = SetAnyPropertyToolPreProcessor.class)
    @Select(preprocessed = true)
    private String name;

    @Input(label = "Новое значение")
    private String value;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
}
