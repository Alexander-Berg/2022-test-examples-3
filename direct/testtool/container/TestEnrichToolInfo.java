package ru.yandex.direct.internaltools.tools.testtool.container;

import javax.annotation.ParametersAreNonnullByDefault;

import com.fasterxml.jackson.annotation.JsonProperty;

import ru.yandex.direct.internaltools.core.annotations.output.Enrich;
import ru.yandex.direct.internaltools.core.enums.InternalToolDetailKey;

@ParametersAreNonnullByDefault
public class TestEnrichToolInfo {
    @JsonProperty("BooleanYesNoFetcher")
    @Enrich(InternalToolDetailKey.BOOLEAN_REPLACE)
    private Boolean booleanValue = Boolean.TRUE;

    @JsonProperty("IpWhoisLinkFetcher")
    @Enrich(InternalToolDetailKey.IP_WHOIS_LINK_REPLACE)
    private String ip = "1.1.1.1";

    @JsonProperty("UserDataByLoginFetcher")
    @Enrich(InternalToolDetailKey.LOGIN)
    private String login = "spb-tester";

    @JsonProperty("CommonLinkFetcher")
    @Enrich(InternalToolDetailKey.COMMON_LINK_REPLACE)
    private String commonUrl = "https://st.yandex-team.ru/DIRECT-99895";

    public TestEnrichToolInfo withBooleanValue(Boolean booleanValue) {
        this.booleanValue = booleanValue;
        return this;
    }

    public TestEnrichToolInfo withIp(String ip) {
        this.ip = ip;
        return this;
    }

    public TestEnrichToolInfo withLogin(String login) {
        this.login = login;
        return this;
    }

    public TestEnrichToolInfo withCommonUrl(String commonUrl) {
        this.commonUrl = commonUrl;
        return this;
    }

    public Boolean getBooleanValue() {
        return booleanValue;
    }

    public String getIp() {
        return ip;
    }

    public String getLogin() {
        return login;
    }

    public String getCommonUrl() {
        return commonUrl;
    }
}
