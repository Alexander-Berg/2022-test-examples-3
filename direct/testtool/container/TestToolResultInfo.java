package ru.yandex.direct.internaltools.tools.testtool.container;

import javax.annotation.ParametersAreNonnullByDefault;

@ParametersAreNonnullByDefault
public class TestToolResultInfo {
    private String key;
    private String value;

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public TestToolResultInfo withKey(String key) {
        this.key = key;
        return this;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public TestToolResultInfo withValue(String value) {
        this.value = value;
        return this;
    }
}
