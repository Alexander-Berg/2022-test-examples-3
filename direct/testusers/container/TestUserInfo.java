package ru.yandex.direct.internaltools.tools.testusers.container;

import javax.annotation.ParametersAreNonnullByDefault;

import com.fasterxml.jackson.annotation.JsonProperty;

import ru.yandex.direct.rbac.RbacRole;

@ParametersAreNonnullByDefault
public class TestUserInfo {

    @JsonProperty("Логин")
    private String login;

    @JsonProperty("Доменный логин")
    private String domainLogin;

    @JsonProperty("Роль")
    private RbacRole role;

    public String getLogin() {
        return login;
    }

    public String getDomainLogin() {
        return domainLogin;
    }

    public RbacRole getRole() {
        return role;
    }

    public TestUserInfo withLogin(String login) {
        this.login = login;
        return this;
    }

    public TestUserInfo withDomainLogin(String domainLogin) {
        this.domainLogin = domainLogin;
        return this;
    }

    public TestUserInfo withRole(RbacRole role) {
        this.role = role;
        return this;
    }
}
