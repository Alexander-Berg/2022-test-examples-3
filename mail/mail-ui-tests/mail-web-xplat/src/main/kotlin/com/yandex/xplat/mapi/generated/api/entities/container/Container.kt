// <<< AUTOGENERATED BY YANDEX.SCRIPT FROM api/entities/container/container.ts >>>

package com.yandex.xplat.mapi

import com.yandex.xplat.common.*

public open class Container(val stats: ContainerStats, val folders: YSArray<Folder>, val labels: YSArray<Label>): ResponseWithStatus {
    open override fun networkStatus(): NetworkStatus {
        return this.stats.status
    }

}

public fun containerFromJSONItem(item: JSONItem): Container? {
    if (item.kind != JSONItemKind.array) {
        return null
    }
    val array = item as ArrayJSONItem
    val containerStats = containerStatsFromJSONItem(array.`get`(0))
    if (containerStats == null) {
        return null
    }
    val folders: YSArray<Folder> = mutableListOf()
    val labels: YSArray<Label> = mutableListOf()
    if (containerStats.status.code == NetworkStatusCode.ok) {
        for (value in array.asArray()) {
            val map = value as MapJSONItem
            if (map.hasKey("fid")) {
                folders.add(folderFromJSONItem(map)!!)
            } else if (map.hasKey("lid")) {
                labels.add(labelFromJSONItem(map)!!)
            }
        }
    }
    return Container(containerStats, folders, labels)
}

