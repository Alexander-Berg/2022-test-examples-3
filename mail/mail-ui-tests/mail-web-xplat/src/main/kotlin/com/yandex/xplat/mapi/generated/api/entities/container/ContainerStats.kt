// <<< AUTOGENERATED BY YANDEX.SCRIPT FROM api/entities/container/container-stats.ts >>>

package com.yandex.xplat.mapi

import com.yandex.xplat.common.JSONItem
import com.yandex.xplat.common.JSONItemKind
import com.yandex.xplat.common.MapJSONItem

public open class ContainerStatsPayload(val md5: String, val mailboxRevision: Int) {
}

public open class ContainerStats(val status: NetworkStatus, val payload: ContainerStatsPayload?) {
}

public fun containerStatsFromJSONItem(item: JSONItem): ContainerStats? {
    if (item.kind != JSONItemKind.map) {
        return null
    }
    val map = item as MapJSONItem
    val status = networkStatusFromJSONItem(map.`get`("status")!!)!!
    if (status.code != NetworkStatusCode.ok) {
        return ContainerStats(status, null)
    }
    val md5 = map.getString("md5")!!
    val mailboxRevision = map.getInt32("mailbox_revision")!!
    return ContainerStats(status, ContainerStatsPayload(md5, mailboxRevision))
}

