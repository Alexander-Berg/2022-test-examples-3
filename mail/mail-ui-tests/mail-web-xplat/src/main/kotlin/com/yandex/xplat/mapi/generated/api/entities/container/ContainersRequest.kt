// <<< AUTOGENERATED BY YANDEX.SCRIPT FROM api/entities/container/containers-request.ts >>>

package com.yandex.xplat.mapi

import com.yandex.xplat.common.*

public open class ContainersRequest(private val withTabs: Boolean = false): MailNetworkRequest() {
    open override fun urlExtra(): NetworkUrlExtra {
        val url = super.urlExtra()
        if (this.withTabs) {
            url.putString("withTabs", "1")
        }
        return url
    }

    open override fun version(): NetworkAPIVersions {
        return NetworkAPIVersions.v1
    }

    open override fun method(): NetworkMethod {
        return NetworkMethod.`get`
    }

    open override fun path(): String {
        return "xlist"
    }

    open override fun params(): NetworkParams {
        return MapJSONItem()
    }

    open override fun encoding(): RequestEncoding {
        return UrlRequestEncoding()
    }

}

