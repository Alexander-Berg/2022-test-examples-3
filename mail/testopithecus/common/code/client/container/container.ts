import { Nullable } from '../../../ys/ys'
import { ArrayJSONItem, JSONItem, JSONItemKind, MapJSONItem } from '../../mail/logging/json-types'
import { FolderDTO, folderFromJSONItem } from '../folder/folderDTO'
import { Label, labelFromJSONItem } from '../label/label'
import { NetworkStatusCode } from '../status/network-status'
import { ContainerStats, containerStatsFromJSONItem } from './container-stats'

export class Container {
  public constructor(
    public readonly stats: ContainerStats,
    public readonly folders: readonly FolderDTO[],
    public readonly labels: readonly Label[],
  ) { }
}

export function containerFromJSONItem(item: JSONItem): Nullable<Container> {
  if (item.kind !== JSONItemKind.array) {
    return null
  }
  const array = item as ArrayJSONItem
  const containerStats = containerStatsFromJSONItem(array.get(0))
  if (containerStats === null) {
    return null
  }

  const folders: FolderDTO[] = []
  const labels: Label[] = []
  if (containerStats.status.code === NetworkStatusCode.ok) {
    for (const value of array.asArray()) {
      const map = value as MapJSONItem
      if (map.hasKey('fid')) {
        folders.push(folderFromJSONItem(map)!)
      } else if (map.hasKey('lid')) {
        labels.push(labelFromJSONItem(map)!)
      }
    }
  }

  return new Container(containerStats, folders, labels)
}
