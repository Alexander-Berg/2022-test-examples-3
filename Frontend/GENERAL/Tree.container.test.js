import React from 'react';
import { mount } from 'enzyme';

import { TreeContainer } from './Tree.container';

jest.mock('./Tree');
jest.mock('~/src/common/components/WikiFormatter');

describe('Should fetch data', () => {
    it('upon init, should not dispatch fetch action if url is empty', () => {
        const updateRequest = jest.fn();

        const wrapper = mount(
            <TreeContainer
                requestUpdate={updateRequest}
                requestAllNodeChildren={jest.fn()}
                requestNodeChildrenUpdate={jest.fn()}
                setNodeOpened={jest.fn()}
                setNodeClosed={jest.fn()}
                filters={{}}
                tree={{}}
                error={null}
                status="inited"
                rootServiceId={0}
                isUrlEmpty
            />
        );

        expect(updateRequest).not.toHaveBeenCalled();

        wrapper.unmount();
    });

    it('upon init, should dispatch fetch action if url is not empty', () => {
        const updateRequest = jest.fn();

        const wrapper = mount(
            <TreeContainer
                requestUpdate={updateRequest}
                requestAllNodeChildren={jest.fn()}
                requestNodeChildrenUpdate={jest.fn()}
                setNodeOpened={jest.fn()}
                setNodeClosed={jest.fn()}
                filters={{}}
                tree={new Map()}
                error={null}
                status="inited"
                rootServiceId={0}
                isUrlEmpty={false}
            />
        );

        expect(updateRequest).toHaveBeenCalled();

        wrapper.unmount();
    });

    it('upon request filters update, debounced 1000ms', done => {
        const updateRequest = jest.fn();

        const wrapper = mount(
            <TreeContainer
                requestUpdate={updateRequest}
                requestAllNodeChildren={jest.fn()}
                requestNodeChildrenUpdate={jest.fn()}
                setNodeOpened={jest.fn()}
                setNodeClosed={jest.fn()}
                filters={{}}
                tree={{}}
                error={null}
                status="inited"
                rootServiceId={0}
            />
        );

        const initialCalls = updateRequest.mock.calls.length;

        wrapper.setProps({ filters: { foo: 'bar' } });

        // fake timers не работают из коробки с lodash debounce, нужно мокать чуть сложнее
        // https://github.com/facebook/jest/issues/3465#issuecomment-504908570
        setTimeout(() => {
            expect(updateRequest).toHaveBeenCalledTimes(initialCalls + 1);
            wrapper.unmount();
            done();
        }, 1000);
    });
});

describe('Should prepare data', () => {
    it('for the view component', () => {
        const wrapper = mount(
            <TreeContainer
                requestUpdate={jest.fn()}
                requestAllNodeChildren={jest.fn()}
                requestNodeChildrenUpdate={jest.fn()}
                setNodeOpened={jest.fn()}
                setNodeClosed={jest.fn()}
                isUrlEmpty
                tree={{
                    '0': {
                        isRoot: true,
                        children: [1, 2],
                    },
                    '1': {
                        data: { id: 1, parent: null },
                        childrenDataStatus: 'pending',
                        childrenError: null,
                        isOpen: false,
                        children: [1.1],
                    },
                    '2': {
                        data: { id: 2, parent: null },
                        childrenDataStatus: 'inited',
                        childrenError: null,
                        isOpen: true,
                        children: [2.1, 2.2],
                    },
                    '1.1': {
                        data: { id: 1.1, parent: { id: 1 } },
                        childrenDataStatus: 'pending',
                        childrenError: null,
                        isOpen: false,
                        children: [],
                    },
                    '2.1': {
                        data: { id: 2.1, parent: { id: 2 } },
                        childrenDataStatus: 'pending',
                        childrenError: null,
                        isOpen: false,
                        children: [],
                    },
                    '2.2': {
                        data: { id: 2.2, parent: { id: 2 } },
                        childrenDataStatus: 'pending',
                        childrenError: null,
                        isOpen: false,
                        children: [],
                    },
                }}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
