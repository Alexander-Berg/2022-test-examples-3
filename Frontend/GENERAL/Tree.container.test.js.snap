// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should prepare data for the view component 1`] = `
<TreeContainer
  isUrlEmpty={true}
  requestAllNodeChildren={[MockFunction]}
  requestNodeChildrenUpdate={[MockFunction]}
  requestUpdate={[MockFunction]}
  rootServiceId={0}
  setNodeClosed={[MockFunction]}
  setNodeOpened={[MockFunction]}
  tree={
    Object {
      "0": Object {
        "children": Array [
          1,
          2,
        ],
        "isRoot": true,
      },
      "1": Object {
        "children": Array [
          1.1,
        ],
        "childrenDataStatus": "pending",
        "childrenError": null,
        "data": Object {
          "id": 1,
          "parent": null,
        },
        "isOpen": false,
      },
      "1.1": Object {
        "children": Array [],
        "childrenDataStatus": "pending",
        "childrenError": null,
        "data": Object {
          "id": 1.1,
          "parent": Object {
            "id": 1,
          },
        },
        "isOpen": false,
      },
      "2": Object {
        "children": Array [
          2.1,
          2.2,
        ],
        "childrenDataStatus": "inited",
        "childrenError": null,
        "data": Object {
          "id": 2,
          "parent": null,
        },
        "isOpen": true,
      },
      "2.1": Object {
        "children": Array [],
        "childrenDataStatus": "pending",
        "childrenError": null,
        "data": Object {
          "id": 2.1,
          "parent": Object {
            "id": 2,
          },
        },
        "isOpen": false,
      },
      "2.2": Object {
        "children": Array [],
        "childrenDataStatus": "pending",
        "childrenError": null,
        "data": Object {
          "id": 2.2,
          "parent": Object {
            "id": 2,
          },
        },
        "isOpen": false,
      },
    }
  }
>
  <mockConstructor
    error={null}
    list={
      Array [
        Object {
          "childrenDataStatus": "pending",
          "childrenError": null,
          "data": Object {
            "id": 1,
            "parent": null,
          },
          "displayedChildrenCount": 1,
          "isOpen": false,
        },
        Object {
          "childrenDataStatus": "inited",
          "childrenError": null,
          "data": Object {
            "id": 2,
            "parent": null,
          },
          "displayedChildrenCount": 2,
          "isOpen": true,
        },
        Object {
          "childrenDataStatus": "pending",
          "childrenError": null,
          "data": Object {
            "id": 2.1,
            "parent": Object {
              "id": 2,
            },
          },
          "displayedChildrenCount": 0,
          "isOpen": false,
        },
        Object {
          "childrenDataStatus": "pending",
          "childrenError": null,
          "data": Object {
            "id": 2.2,
            "parent": Object {
              "id": 2,
            },
          },
          "displayedChildrenCount": 0,
          "isOpen": false,
        },
      ]
    }
    onNodeClick={[Function]}
    onShowUnmatched={[Function]}
  >
    <div>
      Tree mock. Props used: 
      {"list":[{"data":{"id":1,"parent":null},"childrenDataStatus":"pending","childrenError":null,"displayedChildrenCount":1,"isOpen":false},{"data":{"id":2,"parent":null},"childrenDataStatus":"inited","childrenError":null,"displayedChildrenCount":2,"isOpen":true},{"data":{"id":2.1,"parent":{"id":2}},"childrenDataStatus":"pending","childrenError":null,"displayedChildrenCount":0,"isOpen":false},{"data":{"id":2.2,"parent":{"id":2}},"childrenDataStatus":"pending","childrenError":null,"displayedChildrenCount":0,"isOpen":false}],"error":null}
    </div>
  </mockConstructor>
</TreeContainer>
`;
